src/tools/calculateChanges.ts

```diff
@@ -0,0 +1,207 @@
+            // For new and modified properties of the object field, we can just pass them through to
+            // APIv2.
+            if (currentAssetState[fieldName]) {
+                // eslint-disable-next-line no-restricted-syntax
+                for (const currentObjectFieldPropertyName of Object.keys(
+                    currentAssetState[fieldName]
+                )) {
+                    // Loop through each object property that exists currently
+                    if (desiredAssetState[fieldName][currentObjectFieldPropertyName] === undefined) {
+                        // If the object has a property that exists now but doesn't exist in the proposed state
+                        if (allowDelete) {
+                            desiredAssetStateWithChanges[fieldName][
+                                currentObjectFieldPropertyName
+                            ] = null;
+                        } else {
+                            // If deletes aren't allowed, do outright delete the property within the object
+                            log.warn(
+                                `Detected that the ${fieldName} of the following ${handler.name || handler.id || ''
+                                } should be deleted. Doing so may be destructive.\nYou can enable deletes by setting 'AUTH0_ALLOW_DELETE' to true in the config\n${handler.objString(
+                                    currentAssetState
+                                )}`
+                            );
+                        }
+                    }
+                }
+            }
+        } else if (allowDelete) {
+            // If the desired state does not have the object field and the current state does, we
+            // should mark *all* properties for deletion by specifying an empty object.
+            //
+            // See: https://auth0.com/docs/users/metadata/manage-metadata-api#delete-user-metadata
+            desiredAssetStateWithChanges[fieldName] = {};
+        } else {
+            delete desiredAssetStateWithChanges[fieldName];
+            log.warn(
+                `Detected that the ${fieldName} of the following ${handler.name || handler.id || ''
+                } should be emptied. Doing so may be destructive.\nYou can enable deletes by setting 'AUTH0_ALLOW_DELETE' to true in the config\n${handler.objString(
+                    currentAssetState
+                )}`
+            );
+        }
+    }
+
+    return desiredAssetStateWithChanges;
+}
+
+// Temporary type for Asset until more canonical types are created
+type Asset = {
+    id?: string
+    name?: string
+    [key: string]: { [key: string]: string } | string//Temporarily accounting for different types of assets with arbitrary properties
+}
+
+// Temporary type for Auth0 API handler until more canonical types are created
+type Handler = {
+    id: string
+    name?: string
+    objectFields: string[]
+    objString: (arg0: Asset) => string
+
+}
```

Comment:
Temporarily defining these types within this file until more canonical types are generated from their respective TS migrations.